python/ray/_private/worker.py

```diff
@@ -2343,29 +2343,38 @@ def connect(
 
     if mode == SCRIPT_MODE:
         # Add the directory containing the script that is running to the Python
-        # paths of the workers. Also add the current directory. Note that this
-        # assumes that the directory structures on the machines in the clusters
-        # are the same.
-        # When using an interactive shell, there is no script directory.
-        # We also want to skip adding script directory when running from dashboard.
-        code_paths = []
-        if not interactive_mode and not (
-            namespace and namespace == ray_constants.RAY_INTERNAL_DASHBOARD_NAMESPACE
+        # paths of the workers. Also add the current directory.
+        #
+        # Note that this is only meaningful when all of these are true:
+        # (1) there's no working_dir (or code search path should be in working_dir),
+        # (2) it's not interactive mode, (there's no script file in interactive mode),
+        # (3) it's not in dashboard,
+        # (4) it's not client mode, (handled by client code)
+        # (5) the driver is at the same node (machine) as the worker.
+        #
+        # We only do the first 4 checks here.
+        # TODO: do the (5) check by also passing node id.
```

Comment:
why don't we just do it here?

Reply:
We want to compare node_id of the driver vs node_id of this worker. We don't have the former value IIUC.

Reply:
Hmm I prefer to just fix 100% in this PR. Isn't this pretty easy to do it? (just add driver node id to job config?)